crates/accelerate/src/commutation_cancellation.rs

```diff
@@ -0,0 +1,280 @@
+                || matches!(cancel_key.gate, GateOrRotation::XRotation)
+            {
+                let run_qarg = match &dag.dag[*cancel_set.first().unwrap()] {
+                    NodeType::Operation(instr) => dag.get_qargs(instr.qubits)[0],
+                    _ => panic!("Unexpected type in commutation set run."),
+                };
+                let mut total_angle: f64 = 0.0;
+                let mut total_phase: f64 = 0.0;
+                for current_node in cancel_set {
+                    let node_op = match &dag.dag[*current_node] {
+                        NodeType::Operation(instr) => instr,
+                        _ => panic!("Unexpected type in commutation set run."),
+                    };
+                    let node_op_name = node_op.op.name();
+
+                    let node_qargs = dag.get_qargs(node_op.qubits);
+                    if node_op
+                        .extra_attrs
+                        .as_deref()
+                        .is_some_and(|attr| attr.condition.is_some())
+                        || node_qargs.len() > 1
+                        || node_qargs[0] != run_qarg
+                    {
+                        return Err(QiskitError::new_err("internal error"));
+                    }
+
+                    let node_angle = if ROTATION_GATES.contains(&node_op_name) {
+                        match node_op.params_view().first() {
+                            Some(Param::Float(f)) => Ok(*f),
+                            _ => return Err(QiskitError::new_err(format!(
+                                "Rotational gate with parameter expression encountered in cancellation {:?}",
+                                node_op.op
+                            )))
+                        }
+                    } else if HALF_TURNS.contains(&node_op_name) {
+                        Ok(PI)
+                    } else if QUARTER_TURNS.contains(&node_op_name) {
+                        Ok(PI / 2.0)
+                    } else if EIGHTH_TURNS.contains(&node_op_name) {
+                        Ok(PI / 4.0)
+                    } else {
+                        Err(QiskitError::new_err(format!(
+                            "Angle for operation {} is not defined",
+                            node_op_name
+                        )))
+                    };
+                    total_angle += node_angle?;
+
+                    let Param::Float(new_phase) = node_op.op.definition(node_op.params_view()).unwrap().global_phase().clone() else { unreachable!() };
+                    total_phase += new_phase
+                }
+
+                let new_op = if matches!(cancel_key.gate, GateOrRotation::ZRotation) {
+                    z_var_gate.unwrap()
+                } else if matches!(cancel_key.gate, GateOrRotation::XRotation) {
+                    &RXGate
+                } else {
+                    return Err(QiskitError::new_err("impossible case!"));
+                };
```

Comment:
I think using a match here would be more natural:
```suggestion
                let new_op = match cancel_key.gate
                    GateOrRotation::ZRotation => z_var_gate.unwrap(),
                    GateOrRotation::XRotation) => &RXGate,
                    _ => unreachable!()
                };
```

I also switched this to an `unreachable!()` because while reviewing the current form of the PR it is obvious that this branch is never reachable because  after re-reading the code I realize the else arm here couldn't ever be reached since this all in a block that gate matches either rotation variant of the enum.